pkg/tasks/tasks.go

```diff
@@ -147,6 +147,15 @@ func (self *ViewBufferManager) NewCmdTask(start func() (*exec.Cmd, io.Reader), p
 		scanner := bufio.NewScanner(r)
 		scanner.Split(bufio.ScanLines)
 
+		data := make(chan []byte)
+
+		go utils.Safe(func() {
```

Comment:
Is there a risk that we'll get a bunch of these goroutines banked up after iterating through a bunch of items in a side view? Like, do we eventually make it out of the `scanner` loop here? If not, it would be good if there was a way to force us to return from the goroutine.

Although in saying that, if that were possible we'd be able to solve it directly in the original code, hmm...

Reply:
> Is there a risk that we'll get a bunch of these goroutines banked up after iterating through a bunch of items in a side view? 

I just updated the PR description, could you please read and tell me if it makes sense. I also had this pre-conception initially and even wrote more code to deal with it but the testing shows that an old gorouting is killed before a new one is requested so it seems they will never back up. There _could be_ more than one as my testing shows but they all will be stopped one by one eventually, as each new one stops the previous one.

Reply:
> Although in saying that, if that were possible we'd be able to solve it directly in the original code, hmm...

This is entirely up to you, but I think there is still value in merging this:

- it seems to resolve the issue
- it can act as a backstop if there are any further issue with the other resolution
- it could take some time to do the other resolution, while merging this will unblock people waiting on this long standing issue

Reply:
> Like, do we eventually make it out of the scanner loop here?

Oh sorry, it took some time to sink in. So what I believe happens here, is that on `break outer` we end up with `kill` and the kill get scanner unblocked so the goroutine is let go.

Reply:
Sorry I have a habit of editing my comment as my thinking evolves I edited some of the above. As you mention in the issue thread in theory we could change the scanner to a reader to avoid the blocking altogether, but I would need help with this, as it seems (by looking at the scanner source code) quite complicated. Without doing that and without *the other fix* I do not think this could be easily resolved. I aimed for something low effort that would allow people to move on.

Reply:
No need to apologise :) I'm leaning towards going with this approach to fix the immediate problem and then looking for a better fix. It's better to have a memory leak than for nothing to work at all.

My main hesitation is around performance: I'm not sure what effect it'll have to have each line passed to a separate goroutine (for osx/linux where there's no issue atm)

Reply:
@jesseduffield again, possibly dumb question, but where each line is passed to a separate goroutine? As far as I can see we added one go routine in total the one that hold the loop with the scanner. Are we creating a goroutine for every line anywhere?

You may be referring for the lines on the left of lazygit UI, but this part we have not changed we always created a goroutine per line as user press up and down arrows. Can you talk a bit more about your hesitation?

Reply:
If we look at go routine creations it this function the new one is one of four. It has probably the least effect on performance of the four because reading a scanner is not very resource intensive operation.

Reply:
To be clear, I'm not worried about spawning too many goroutines, I'm worried about each line of the command's output needing to go through a channel, such that the go runtime needs to hand off control from the scanner goroutine to the other goroutine many times. This is pure speculation of course, and I'll do some profiling to test the hypothesis now